pkg/scale/decode.go

```diff
@@ -121,46 +119,94 @@ func (ds *decodeState) unmarshal(dstv reflect.Value) (err error) {
 	switch in.(type) {
 	case *big.Int:
 		err = ds.decodeBigInt(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding big integer: %w", err)
+		}
 	case *Uint128:
 		err = ds.decodeUint128(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding Uint128: %w", err)
+		}
 	case int, uint:
 		err = ds.decodeUint(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding uint: %w", err)
+		}
 	case int8, uint8, int16, uint16, int32, uint32, int64, uint64:
 		err = ds.decodeFixedWidthInt(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding fixed width int: %w", err)
+		}
 	case []byte:
 		err = ds.decodeBytes(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding []byte: %w", err)
+		}
 	case string:
 		err = ds.decodeBytes(dstv)
```

Comment:
Are we sure this is correct? the case `[]byte` uses `decodeBytes` as well? 🤔

Reply:
It seems right, it's passing unit tests, and I didn't change this and we're not seeing any issues.

Reply:
Alrighty then, that was just curiously strange. Thanks!